lib/services/security/service.js

```diff
@@ -774,10 +774,17 @@ function tokenFromRequest(req, options) {
   if (!options) options = {};
 
   if (!options.tokenName) options.tokenName = 'happn_token';
-  if (!options.cookieName && this.config.cookieName) options.cookieName = this.config.cookieName;
-  if (!options.cookieName) options.cookieName = 'happn_token';
-  let token = req.cookies.get(options.cookieName);
+  if (!options.cookieName) options.cookieName = this.config.cookieName || 'happn_token';
 
+  let token = req.cookies.get(
+    req.headers['x-forwarded-proto'] === 'https' || req.connection.encrypted
+      ? `${options.cookieName}_https`
+      : options.cookieName
+  );
+  if (token) return token;
+
+  //fall back to the old cookie name, for backward compatibility - old browsers

```

Comment:
Do we need to do this, I can see old unrefreshed browsers breaking if we don't (once they all refresh we should be ok)

Reply:
I think we can leave this out unless someone else feels strong about it.

Reply:
All that will happen is that they will have to login again?

Reply:
I think we should just leave it in unless you guys see a security issue there?